File: PulsarCraft/src/main/java/safrain/nodelet/game/device/invoker/Invoker.js
var ChargeInvoker = newBuilder({
	javaType : 'safrain.nodelet.game.device.invoker.ChargeInvoker',
	filler : function(b, p) {
		b.maxCharge = value(p.maxCharge, 100);
		b.chargeRate = value(p.chargeRate, 0);
		b.currentCharge = value(p.currentCharge, 0);
		b.consume = value(p.consume, 100);
	}
});
var PassiveInvoker = newBuilder({
	javaType : 'safrain.nodelet.game.device.invoker.PassiveInvoker',
	filler : function(b, p) {
		b.canInvoke = value(p.canInvoke, false);
	}
});

File: PulsarCraft/src/main/java/safrain/nodelet/game/device/invoker/Invoker.java
package safrain.nodelet.game.device.invoker;

import safrain.nodelet.factory.BaseBuilder;

public abstract class Invoker {
	public abstract boolean canInvoke();

	public abstract void invoke();

	public void tick() {
	}

	public static abstract class Builder<T extends Invoker> extends
			BaseBuilder<T> {

	}
}
